mono/mini/aot-compiler.c

```diff
@@ -9964,12 +9964,12 @@ compile_asm (MonoAotCompile *acfg)
 		wrap_path (g_strdup_printf ("%s.o", acfg->tmpfname)), ld_flags);
 #else
 	// Default (linux)
-	char *args = g_strdup_printf ("%s %s -shared -o %s %s %s %s", tool_prefix, LD_OPTIONS,
+	char *args = g_strdup_printf ("%s -shared -o %s %s %s %s", LD_OPTIONS,
 		wrap_path (tmp_outfile_name), wrap_path (llvm_ofile),
 		wrap_path (g_strdup_printf ("%s.o", acfg->tmpfname)), ld_flags);
 
 	if (acfg->llvm) {
-		command = g_strdup_printf ("clang++ %s", args);
+		command = g_strdup_printf ("clang++ %s %s", tool_prefix, args);
 	} else {
```

Comment:
not exactly sure about the llvm case, because actually this fixed for me the other case. I don't get why we want to use just `clang++` and ignore `tool_prefix` at all? as it is, it's obviously wrong, so it has to be either:
```c
g_strdup_printf ("clang++ %s", args);
// or
g_strdup_printf ("\"%sclang++\" %s", tool_prefix, args);
```
?